rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubric.js

```diff
@@ -162,6 +200,94 @@ export class SakaiRubric extends RubricsElement {
     this.updateRubric();
   }
 
+  handleCriterionCreated(e) {
+    var data = e.detail;
+    this.rubric.criterions.push(data.criterion);
+  }
+
+  handleSaveWeights(e) {
+    var _this = this;
+    $(`[rubric-id=${this.rubric.id}]`).find('.save-weights').prop('disabled', true);
+    this.rubric.criterions.forEach(cr => {
+      cr.weight = parseFloat(cr.weight);
+      $.ajax({
+        url: `/rubrics-service/rest/criterions/${cr.id}`,
+        headers: { "authorization": this.token },
+        method: "PATCH",
+        contentType: "application/json",
+        data: JSON.stringify({ weight: cr.weight })
+      }).done(data => {
+        $(`[rubric-id=${_this.rubric.id}]`).find('.save-success').fadeIn('100');
+        setTimeout(function () {
+          $(`[rubric-id=${_this.rubric.id}]`).find('.save-weights').prop('disabled', false);
+        }, 1000);
```

Comment:
could you not use this.updateComplete.then( () => here instead of these timeouts?

Reply:
I think that would make it appear/disappear instantly, instead of having a wait time so the user can see the success message.